### PR TITLE
generate: use token groups for delimited token streams

### PR DESCRIPTION
**Pull request by rmsyn**
Uses `proc_macro2::Group` for token streams delimited with braces.

Resolves: #863

### CODE REVIEW

**Review by burrbull — APPROVED** [review]

**Comment by burrbull** [comment]
Looks like need to update some dependencies in `ci/scripts.sh`

**Review by jannic — DISMISSED** [review]
Some parts could be slightly simplified, but the more complex restructuring of `render_register_mod_debug` is exactly as I envisioned it.

**Comment by jannic** [comment]
> What's the best workaround? Pinning an older version of nightly in ci, for now?

If that's acceptable, feel free to copy from https://github.com/rust-embedded/svd2rust/pull/865/commits/c577438574a6fe51cea249113f2b101117173b88

(I didn't bisect the exact version, but just took one that succeeded in a previous CI run.)

File: src/generate/register.rs [hunk 1]
```diff
@@ -136,11 +136,12 @@ pub fn render(
...
             #[doc = #description]
-            pub mod #mod_ty {
-                #mod_items
-            }
+            pub mod #mod_ty
         });
 
+        let mod_group = Group::new(Delimiter::Brace, mod_items);
+        out.extend(quote! { #mod_group });
+
```
[the remark this answers is not in the document]
Comment (rmsyn) [reply src/generate/register.rs:144]: Thanks, I forgot to revert this change while I was still working things out. I'll change it back.

**Comment by rmsyn** [comment]
> CI results at https://github.com/rust-embedded/svd2rust/pull/865

I can keep an eye on #865, and rebase on top of those changes once merged.

Let me know what you would like me to do.

**Comment by burrbull** [comment]
> Let me know what you would like me to do.

Merge #865 changes in this PR, please.

**Review by burrbull — APPROVED** [review]

**Pull request merged**